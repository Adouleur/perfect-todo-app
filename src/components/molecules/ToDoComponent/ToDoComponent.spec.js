import { describe, it, expect, vi, beforeEach } from 'vitest';
import { mount } from '@vue/test-utils';
import ToDo from '@/components/molecules/ToDoComponent/ToDoComponent.vue';
import { createStore } from 'vuex';

// Mock Vuex store
const mockDispatch = vi.fn();
const store = createStore({
  state: {},
  actions: {
    toggleComplete: mockDispatch
  }
});

describe('ToDo Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ToDo, {
      global: {
        plugins: [store]
      },
      props: { text: 'Sample', completed: false, id: 1 }
    });
  });

  it('renders the text prop correctly', () => {
    expect(wrapper.text()).toContain('Sample');
  });

  it('renders a CFormCheck checkbox', () => {
    const checkbox = wrapper.findComponent({ name: 'CFormCheck' });
    expect(checkbox.exists()).toBe(true);
  });

  it('renders a CCloseButton button', () => {
    const closeButton = wrapper.findComponent({ name: 'CCloseButton' });
    expect(closeButton.exists()).toBe(true);
  });

  it('contains a checkbox with correct id', () => {
    const checkbox = wrapper.find('#flexCheckChecked');
    expect(checkbox.exists()).toBe(true);
  });

  it('applies completedTask class when completed prop is true', async () => {
    await wrapper.setProps({ completed: true });
    const span = wrapper.find('span');
    expect(span.classes()).toContain('completedTask');
  });

  it('does not apply completedTask class when completed prop is false', async () => {
    await wrapper.setProps({ completed: false });
    const span = wrapper.find('span');
    expect(span.classes()).not.toContain('completedTask');
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
