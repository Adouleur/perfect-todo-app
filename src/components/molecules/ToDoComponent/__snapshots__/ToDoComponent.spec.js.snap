// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToDo Component > matches the snapshot 1`] = `
"<div data-v-3fcbac7f="" class="toDo" data-cy="task-item" draggable="true">
  <div data-v-3fcbac7f="">
    <!-- Add data-cy to the checkbox --><input data-v-3fcbac7f="" data-cy="complete-checkbox" class="form-check-input" id="flexCheckChecked" type="checkbox"><span data-v-3fcbac7f="" class="" data-cy="task-text">Sample</span>
  </div><button data-v-3fcbac7f="" type="button" class="btn btn-close delete-icon" aria-label="Close" id="closeButton" data-cy="delete-button"></button>
</div>"
`;
